apps/modernization-api/src/main/java/gov/cdc/nbs/service/EventService.java

```diff
@@ -793,6 +826,21 @@ public Page<Investigation> findOpenInvestigationsForPatient(Long patientId, Grap
                 .build();
         return performSearch(query, Investigation.class);
     }
+    
+	public List<Observation> findMorbidityReportsForPatient(Long patientId) {
+		List<BigInteger> subjectEntityIds = new ArrayList<BigInteger>();
+		List<Long> actIds = new ArrayList<Long>();
+		List<Object[]> results = personReposity.getPersonIdsByPersonParentId(patientId);
+		for (Object[] obj : results) {
+			subjectEntityIds.add((BigInteger) obj[0]);
+		}
+		List<Object[]> actIdResults = participationRepository
+				.findIdActUidByIdTypeCdAndIdSubjectEntityUidIn(Constants.REPORT_TYPE, subjectEntityIds);
+		for (Object[] objAct : actIdResults) {
+			actIds.add(((BigInteger) objAct[0]).longValue());
+		}
```

Comment:
Once the repo is updated to return a Long, this can be simplified to not have to handle the Object[]'s

Reply:
@mpeels  updated. Thanks.